serializers/src/main/java/io/pravega/schemaregistry/codec/Codecs.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.codec;
+
+import com.fasterxml.jackson.databind.util.ByteBufferBackedInputStream;
+import io.pravega.schemaregistry.contract.data.CodecType;
+import lombok.Getter;
+import org.apache.commons.io.IOUtils;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for creating codecs for none, snappy or gzip. 
+ */
+public enum Codecs {
```

Comment:
What's the process of adding new codecs? Do we have to come and edit this class? Could we do anything pluggable instead?

Reply:
Users can supply their own custom codecs implementation.
We provide a small library of codecs for gzip and snappy. 

We can add more codecs in future by editing this class.

Reply:
For example, look at this sample which defines its custom encryption codec:
https://github.com/pravega/schema-registry/blob/for-test/samples/src/main/java/io/pravega/schemaregistry/samples/demo/encoding/EncryptionDemo.java

Reply:
I have not made the serializers pluggable.
For serializers, users need to explicitly pick one codec and tell the serializer to use it. 

Conceptually we could have allowed for automatic loading of decoders into a deserializer using ServiceLoaders. 
And that is something we can do in future.